### DISCUSSION TITLE
Tentang Briva Online

### DISCUSSION THREAD

**Discussion by seisoci** [otnansirk/php-snap-bi#6] [Q&A]
kak mau tnya ini, sya kan sedang ingin develop Briva Online yg Inqury dan Payment. disini saya ingin berarti untuk Briva Online ini Inqury dan Paymentnya di hint oleh pihak bank dan kita berikan responsenya betul begitu bkn?

**Comment by otnansirk** [comment]
Halo @seisoci  salam kenal, Maaf saya blum menangkap konteknya. 

Confirm :
1. Apa briva online = BRI Virtual Account Online ?
2. Apa maksudnya ingin integrasi dengan BRI Va Online

boleh dijelaskan case kebutuhannya sperti apa ?

**Reply by seisoci** [reply]
iia BRI Virtual Account Online,
yg saya tahu utk Virtual Account Online itu kita yang menangkap hint request dri Pihak bank lalu server kita yang membuat responsenya sesuai dokumentasi dari masing2 bank. jdi bukan kita yang membuat request Inquiry & Payment ke server banknya.
https://developers.bri.co.id/id/snap-bi/apidocs-virtual-account-briva-online-snap-bi
dari gambar sequence diagramnya sih gtu mas. tpi sya cek dari code Trait BCA Virtual Account yang dibuat oleh kk, kk yang request ke bank Inquiry & Paymentnya. mohon maaf jika salah

**Reply by otnansirk** [reply]
Benar. Briva akan hit API kita tapi sebelum itu ada proses authentication mas. untuk dapatkan accessToken yang akan digunakan untuk proses inquiry

**Reply by otnansirk** [reply]
Jadi customer perlu tetep akses Backend API kita dulu, nanti backend api melkukan inquiry ke BRIva. 
Setelah ini Briva akan hit api kita (yang sudah di daftarkan ke briva).  jika ada perubahan status pembayaran.

**Reply by seisoci** [reply]
jdi kasus sya ini mas, saya ingin mengajukan pendaftaran BRI VA dan diberikan 2 opsi:
1. BRIVA ONLINE
2. BRIVA WS
dan saya memilih opsi pertama, dan saya diharuskan membuat token, inquiry, dan payment nya. berarti mereka nantinya akan request inquiry ke api yang kita buat dan memberikan response data tagihan costumer kami. yang saya bingung kenapa customer harus akses backend api kita dlu sedangkan kita hanya menunggu response inquiry dari pihak bank

**Reply by otnansirk** [reply]
Sebelumnya sudah ada info dari pihak BRI nya mas ? terkait flow BRIVA Online ?


kalau kita lihat dari flow di documentationnya. 
1. customer request bri itu kan menggunakan API inquiry.
2. sedangkan api inquiry ini membutuhkan authentication dengan access_token. 
3. access_token didapat dari api /access-token/b2b. Dimana ini membutuhkan header X-SIGNATURE. 
4.  X-SIGNATURE di dapat dari generate data di ataranya secret credential yang kita peroleh dari BRI nya mas. 

Jadi hit ke api kita sebenarnya untuk generate access_token terlebih dahulu mas. yang akan di gunakan untuk proses inquiry. 

Kalau di buat flownya jadi seperti ini :
![Screenshot from 2023-10-27 20-34-11](https://github.com/otnansirk/php-snap-bi/assets/30102523/86dbb8fb-d810-468b-b3a3-5c1b68420eab)

yang aku tandai kotak hijau garis putus" itu yang kita di dokumentasi BRI. 

Dan sebelum kesitu ada proses di atasnya untuk get access_token sesuai kebutuhan api inquiry.

**Reply by otnansirk** [reply]
ini bisa di kroscek ke dokumentasi yang mas kirim sebelumnya.

Untuk pernyataan 
```
dan saya diharuskan membuat token, inquiry, dan payment nya.
```

apa datang dari pihak BRI nya mas ?

**Reply by seisoci** [reply]
oke nnti sya tnyakan kembali dgn tim BRI.
btul mas jdi mereka minta sya isi form di excel UAT url token, inquiry dan payment, dan secret code & client id utk UATnya

**Reply by otnansirk** [reply]
Kalau diminta seperti ini berati masnya akan menjadi auth server (penyedia client key dan id. Yang nanti ini digunakan BRI untuk generate akses token dan digunakan untuk send request ke API mas nya.

Kalau ini setahu saya _send-notification_.  kalau kasus yang biasa saya temui di gunakan untuk update data bank-statement.

Tapi baiknya memang di confirmasi lagi ke tim BRI nya mas. 
karena mungkin penanganannya akan berbeda.
